dask/bag/text.py

```diff
@@ -92,6 +92,11 @@ def read_text(urlpath, blocksize=None, compression='infer',
         return from_delayed(blocks)
 
 
+def file_to_blocks(f):
+    with f as f:
+        return list(f)
```

Comment:
This may resolve that problem:

```python
--- a/dask/bag/text.py
+++ b/dask/bag/text.py
@@ -71,7 +71,7 @@ def read_text(urlpath, blocksize=None, compression='infer',
             files = open_text_files(urlpath, encoding=encoding, errors=errors,
                                     compression=compression,
                                     **(storage_options or {}))
-            blocks = [delayed(file_to_blocks)(file) for file in files]
+            blocks = [delayed(list, pure=True)(delayed(file_to_blocks)(file)) for file in files]
 
         else:
             _, blocks = read_bytes(urlpath, delimiter=linedelimiter.encode(),
@@ -94,7 +94,8 @@ def read_text(urlpath, blocksize=None, compression='infer',
 
 def file_to_blocks(f):
     with f as f:
-        return list(f)
+        for line in f:
+            yield line
```

Generally how this works is that we place `list` calls in the graph defensively and then fuse and remove them later on if its safe to do so.  Here is a small example with `count`.


```python
In [1]: import dask.bag as db

In [2]: b = db.read_text('AUTHORS.md', blocksize=None)

In [3]: b.dask
Out[3]: 
{'list-39fdefc70e3f4ef5746e4ab0e678c1e5': (list,
  'file_to_blocks-6ca6c741185277eebf0dc395a17efe7f'),
 ('bag-from-delayed-c32b2a252bee958ed9566d90e0ff33b7',
  0): 'list-39fdefc70e3f4ef5746e4ab0e678c1e5',
 'file_to_blocks-6ca6c741185277eebf0dc395a17efe7f': (<function dask.bag.text.file_to_blocks>,
  <dask.bytes.core.OpenFile at 0x7fb32b0f0940>)}

In [4]: c = b.count()

In [5]: c.dask
Out[5]: 
{'list-39fdefc70e3f4ef5746e4ab0e678c1e5': (list,
  'file_to_blocks-6ca6c741185277eebf0dc395a17efe7f'),
 'count-part-0b62d8a7d985c803f568ea5febb12582': (<function dask.bag.core.empty_safe_apply>,
  <function cytoolz.itertoolz.count>,
  ('bag-from-delayed-c32b2a252bee958ed9566d90e0ff33b7', 0)),
 ('bag-from-delayed-c32b2a252bee958ed9566d90e0ff33b7',
  0): 'list-39fdefc70e3f4ef5746e4ab0e678c1e5',
 'file_to_blocks-6ca6c741185277eebf0dc395a17efe7f': (<function dask.bag.text.file_to_blocks>,
  <dask.bytes.core.OpenFile at 0x7fb32b0f0940>)}

In [6]: c._optimize(c.dask, c._keys())
Out[6]: 
{'count-part-0b62d8a7d985c803f568ea5febb12582': (<function dask.bag.core.empty_safe_apply>,
  <function cytoolz.itertoolz.count>,
  (<function dask.bag.text.file_to_blocks>,
   <dask.bytes.core.OpenFile at 0x7fb32b0f0940>))}
```

You'll notice that while there are extra calls to `list` in the main graph the optimized one is pretty clean.